lib/pluginmanager/sources/base.rb

```diff
@@ -0,0 +1,47 @@
+# encoding: utf-8
+require "rubygems/package"
+
+module LogStash::PluginManager::Sources
+
+  class Base
+
+    attr_reader :uri
+
+    def initialize(uri)
+      @uri = URI(uri)
+    end
+
+    def exist?
+      raise "NotImplemented"
+    end
+
+    def fetch(dest="")
+      raise "NotImplemented"
+    end
+
+    def version
+      File.basename("#{uri}", ".zip").split("-")[1]
+    end
+
+    def to_s
+      "#{uri}"
+    end
+
+    def valid?
+      return false if version.nil? || version.empty?
+      is_zip_file? && valid_version?
+    end
+
+    def valid_version?
+      LOGSTASH_VERSION == version
+    end
```

Comment:
Yeah its a sad reality  :(

On a related note, the only way to enforce a plugin from a pack to work only on specific version of Logstash would be to make it a strict dependency on version of the core.

OK lets move on.